lib/middleware.py

```diff
@@ -75,96 +70,139 @@ def get_rbac_permissions(app, request_header=None):
+# Determine whether the request should be allowed, strictly according to the given permissions.
+# If any of these match, the endpoint should at least be allowed (but may have filtered results).
+def _is_request_allowed_by_permission(
+    rbac_permission: str, permission_base: str, resource_type: RbacResourceType, required_permission: RbacPermission
+) -> bool:
+    return (
+        rbac_permission  # inventory:*:*
+        == f"{permission_base}:{RbacResourceType.ALL.value}:{RbacPermission.ADMIN.value}"
+        or rbac_permission  # inventory:{type}:*
+        == f"{permission_base}:{resource_type.value}:{RbacPermission.ADMIN.value}"
+        or rbac_permission  # inventory:*:(read | write)
+        == f"{permission_base}:{RbacResourceType.ALL.value}:{required_permission.value}"
+        or rbac_permission  # inventory:{type}:(read | write)
+        == f"{permission_base}:{resource_type.value}:{required_permission.value}"
+    )
+
+
+# Validate RBAC response, and fetch
+def _get_group_list_from_resource_definition(resource_definition: dict) -> list[str]:
+    if "attributeFilter" in resource_definition:
+        if resource_definition["attributeFilter"].get("key") != "group.id":
+            abort(
+                HTTPStatus.SERVICE_UNAVAILABLE,
+                "Invalid value for attributeFilter.key in RBAC response.",
+            )
+        elif resource_definition["attributeFilter"].get("operation") != "in":
+            abort(
+                HTTPStatus.SERVICE_UNAVAILABLE,
+                "Invalid value for attributeFilter.operation in RBAC response.",
+            )
+        elif not isinstance(resource_definition["attributeFilter"]["value"], list):
+            abort(
+                HTTPStatus.SERVICE_UNAVAILABLE,
+                "Did not receive a list for attributeFilter.value in RBAC response.",
+            )
+        else:
+            # Validate that all values in the filter are UUIDs.
+            group_list = resource_definition["attributeFilter"]["value"]
+            try:
+                for gid in group_list:
+                    if gid is not None:
+                        UUID(gid)
+            except ValueError:
+                abort(
+                    HTTPStatus.SERVICE_UNAVAILABLE,
+                    "Received invalid UUIDs for attributeFilter.value in RBAC response.",
+                )
+
+            return group_list
+
+
+def get_rbac_filter(
+    resource_type: RbacResourceType,
+    required_permission: RbacPermission,
+    identity: Identity,
+    rbac_request_headers: dict,
+    permission_base: str = "inventory",
+) -> Tuple[str, dict]:
+    # Returns a 2-item tuple:
+    # 1. Whether or not the request should be allowed at all
+    # 2. The filter that should be applied to the data (if allowed)
+    if inventory_config().bypass_rbac:
+        return True, None
+
+    current_identity = get_current_identity()
```

Comment:
I'm afraid it will fail in the export-service. As we are not receiving an API call, the application does not create a new context, so the `connexion.context.context["user"]` attribute will not exist. The app runs once, in a infinite loop, and it leaves the normal API pipeline that Flask creates when receiving API calls

That is the reason we were using `current_identity = create_mock_identity_with_org_id(org_id)` to create the identity, because we needed to mock it every time we received an message from Kafka.

I would suggest validation if the `identity` exists, if not get it using `get_current_identity()`

```suggestion
   if not identity:
       current_identity = get_current_identity()
```

Reply:
My mistake; this was leftover from the original implementation. This line doesn't need to be here. When coming from `rbac()`, the identity is already acquired from the connexion context. When coming from the export service, it's already set to the mock identity.